pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/FilterParser.java

```diff
@@ -356,8 +442,8 @@ private String parseString() throws Exception {
         int i;
 
         // starting from index + 1 to skip leading "
-        for (i = index + 1; i < filterString.length(); ++i) {
-            char chr = filterString.charAt(i);
+        for (i = index + 1; i < filterByteArr.length; ++i) {
+            char chr = (char) filterByteArr[i];
```

Comment:
do we still have quotes around string values ? since we are passing length, it is no longer necessary